hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala

```diff
@@ -426,11 +429,13 @@ trait JSBuilderSanityChecks(instrument: Bool) extends JSBuilder:
   
   val functionParamVarargSymbol = semantics.TempSymbol(0, N, "args")
   
-  override def setupFunction(name: Str, params: List[semantics.Param], body: Block)(using Raise, Scope): (Document, Document) =
+  override def setupFunction(name: Option[Str], params: List[semantics.Param], body: Block)(using Raise, Scope): (Document, Document) =
     if instrument then
       val paramsList = params.map(p => Scope.scope.allocateName(p.sym))
       val paramsStr = Scope.scope.allocateName(functionParamVarargSymbol)
-      val checkArgsNum = doc"if ($paramsStr.length !== ${params.length}) { throw new globalThis.Error('Function \'$name\' expected ' + ${params.length} + ' arguments but got ' + $paramsStr.length) }\n"
+      val fname = name.map(n => s" '${JSBuilder.escapeStringCharacters(n)}' ").getOrElse(" ")
```

Comment:
This should be a `fold`.

Reply:
Thanks! I just changed this `map` and `getOrElse` to `fold` and pushed the commit.